request.go

```diff
@@ -38,7 +42,7 @@ type state struct {
 // New Request initialized based on packet data
 func requestFromPacket(pkt hasPath) *Request {
 	method := requestMethod(pkt)
-	request := NewRequest(method, pkt.getPath())
+	request := newRequestWithContext(method, pkt.getPath())
```

Comment:
Why not leverage the cancelation of a context tree doing the following:

```go
func requestFromPacket(ctx context.Context, pkt hasPath) *Request {
  	method := requestMethod(pkt)
	request := NewRequest(method, pkt.getPath())
	request.ctx, request.cancelCtx = context.WithCancel(ctx)
	// ...
```

Then, change the signature of `packetWorker`

```go
func (rs *RequestServer) packetWorker(ctx context.Context, pktChan chan requestPacket) error {
	// Use ctx when calling requestFromPacket
	// ...
}
```

And finally, `Serve`:

```go
 func (rs *RequestServer) Serve() error {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	var wg sync.WaitGroup
 	runWorker := func(ch requestChan) {
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
 			if err := rs.packetWorker(ctx, ch); err != nil {
 				rs.conn.Close() // shuts down recvPacket
 			}
 		}()
 	}
 	// ...
}
```

Reply:
@ernesto-jimenez Thanks. That sounds like a great idea. Let me make those changes and see them in context, if they look as good as they sound I'll be pushing up an amended PR soon.